docusaurus/docs/dev-docs/plugins/content-manager-apis.md

```diff
@@ -0,0 +1,295 @@
+
+#### ListViewContext
+
+```jsx
+interface ListViewContext {
+  /**
+   * Will be either 'single-types' | 'collection-types'
+   */
+  collectionType: string;
+  /**
+   * The current selected documents in the table
+   */
+  documents: Document[];
+  /**
+   * The current content-type's model.
+   */
+  model: string;
+}
+```
+
+#### EditViewContext
+
+```jsx
+interface EditViewContext {
+  /**
+   * This will only be null if the content-type
+   * does not have draft & publish enabled.
+   */
+  activeTab: 'draft' | 'published' | null;
+  /**
+   * Will be either 'single-types' | 'collection-types'
+   */
+  collectionType: string;
+  /**
+   * Will be undefined if someone is creating an entry.
+   */
+  document?: Document;
+  /**
+   * Will be undefined if someone is creating an entry.
+   */
+  documentId?: string;
+  /**
+   * Will be undefined if someone is creating an entry.
+   */
+  meta?: DocumentMetadata;
+  /**
+   * The current content-type's model.
+   */
+  model: string;
+}
+```
+
+:::tip
+More information about types and APIs can be found in [Strapi's codebase, in the `/admin/src/content-manager.ts` file](https://github.com/strapi/strapi/blob/v5/main/packages/core/content-manager/admin/src/content-manager.ts).
+:::
+
+**Example:**
+
+Adding a panel to the sidebar can be done this way:
+
+```jsx title="my-plugin/components/my-panel.ts"
+import type { PanelComponent, PanelComponentProps } from '@strapi/content-manager/strapi-admin';
+
+const Panel: PanelComponent = ({ 
+  activeTab, 
+  collectionType, 
+  document, 
+  documentId, 
+  meta, 
+  model 
+}: PanelComponentProps) => {
+  return {
+    title: 'My Panel',
+    content: <p>I'm on {activeTab}</p>
+  }
+}
+```
+
+## Available APIs
+
+### `addEditViewSidePanel`
+
+Use this to add new panels to the Edit view sidebar, just like in the following example where something is added to the Releases panel:
+
+![addEditViewSidePanel](/img/assets/content-manager-apis/add-edit-view-side-panel.png)
+
+```jsx
+addEditViewSidePanel(panels: DescriptionReducer<PanelComponent> | PanelComponent[])
+```
+
+#### PanelDescription
+
+The interface of the API only receives the following 2 properties:
+
+```jsx
+{
+  title: string;
+  content: React.ReactNode;	
+}
+```
+
+### `addDocumentAction`
+
+Use this API to add more actions to the Edit view of the Content Manager. There are 3 positions available:
+
+- `header`:
+
+    ![addEditViewSidePanel](/img/assets/content-manager-apis/add-document-header-action.png)
+- `table-row`:
+
+    ![addEditViewSidePanel](/img/assets/content-manager-apis/add-document-action.png)
+- `panel`:
+
+    ![addEditViewSidePanel](/img/assets/content-manager-apis/add-document-action-bis.png)
+
```

Comment:
There is a mistake with this, screenshots should be as following:

header:
![image](https://github.com/user-attachments/assets/39e19191-6864-4de4-8c4a-acbc6e0576f7)


table-row
![Screenshot 2024-09-17 at 16 40 54](https://github.com/user-attachments/assets/67ea4d78-93f7-499d-bb48-81f18adc15e0)

panel:
![image](https://github.com/user-attachments/assets/5544b66e-8e12-4e2a-b2cb-f3e9b58111e4)